src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -355,157 +483,113 @@ public int getDimension()
-    }
-
-    /**
-     * Returns the world the colony is in.
-     *
-     * @return World the colony is in.
-     */
-    @Nullable
-    public World getWorld()
-    {
-        return world;
-    }
-
-    @Override
-    public long getDistanceSquared(@NotNull final BlockPos pos)
-    {
-        return BlockPosUtil.getDistanceSquared2D(center, pos);
-    }
-
-    @Override
-    public boolean hasTownHall()
-    {
-        return townHall != null;
-    }
-
-    /**
-     * Returns the ID of the colony.
+     * Increment the statistic amount and trigger achievement.
      *
-     * @return Colony ID.
-     */
-    @Override
-    public int getID()
-    {
-        return id;
-    }
-
-    /**
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * @param statistic the statistic.
      */
-    public void incrementMobsKilled()
+    public void incrementStatistic(@NotNull String statistic)
     {
-        killedMobs++;
-        final int mobKills = this.getKilledMobs();
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIRST)
+        final int statisticAmount = this.getStatisticAmount(statistic);
+        incrementStatisticAmount(statistic);
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIRST)
         {
-            this.triggerAchievement(ModAchievements.achievementKillOneMob);
+            TriggerColonyAchievements.triggerFirstAchievement(statistic, this);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_SECOND)
+        if (statisticAmount >= NUM_ACHIEVEMENT_SECOND)
         {
-            this.triggerAchievement(ModAchievements.achievementKill25Mobs);
+            TriggerColonyAchievements.triggerSecondAchievement(statistic, this);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_THIRD)
+        if (statisticAmount >= NUM_ACHIEVEMENT_THIRD)
         {
-            this.triggerAchievement(ModAchievements.achievementKill100Mobs);
+            TriggerColonyAchievements.triggerThirdAchievement(statistic, this);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FOURTH)
+        if (statisticAmount >= NUM_ACHIEVEMENT_FOURTH)
         {
-            this.triggerAchievement(ModAchievements.achievementKill500Mobs);
+            TriggerColonyAchievements.triggerFourthAchievement(statistic, this);
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIFTH)
+        if (statisticAmount >= NUM_ACHIEVEMENT_FIFTH)
         {
-            this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
+            TriggerColonyAchievements.triggerFifthAchievement(statistic, this);
         }
     }
 
     /**
-     * get the amount of killed mobs.
+     * Get the amount of statistic.
      *
-     * @return amount of mobs killed
+     * @param statistic the statistic.
+     * @return amount of statistic.
      */
-    public int getKilledMobs()
+    private int getStatisticAmount(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 11, down to the maximum allowed 6. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)